packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -693,6 +668,39 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
     _backgroundColor = widget.items[widget.currentIndex].backgroundColor;
   }
 
+  // Computes the default value for the [type] parameter.
+  //
+  // If type is provided, it is returned. Next, if the bottom navigation bar
+  // theme provides a type, it is used. Finally, the default behavior will be
+  // [BottomNavigationBarType.fixed] for 3 or fewer items, and
+  // [BottomNavigationBarType.shifting] is used for 4+ items.
+  BottomNavigationBarType get _effectiveType {
+
+    if (widget.type != null) {
+      return widget.type;
+    }
+    final BottomNavigationBarType themeType = Theme.of(context).bottomNavigationBarTheme.type;
+    if (themeType != null) {
+      return themeType;
+    }
+    return widget.items.length <= 3 ? BottomNavigationBarType.fixed : BottomNavigationBarType.shifting;
+  }
+
+  // Computes the default value for the [showUnselected] parameter.
+  //
+  // Unselected labels are shown by default for [BottomNavigationBarType.fixed],
+  // and hidden by default for [BottomNavigationBarType.shifting].
+  bool get _defaultShowUnselected {
+    switch (_effectiveType) {
+      case BottomNavigationBarType.shifting:
+        return false;
+      case BottomNavigationBarType.fixed:
+        return true;
+    }
+    assert(false);
```

Comment:
What is this for?

Reply:
I think it's to show that the above switch case should cover all cases, although I did not add this line, it's moved from where this function used to live.